tick/hawkes/simulation/base/simu_point_process.py

```diff
@@ -49,6 +49,9 @@ class SimuPointProcess(Simu):
     intensity_track_step : `float`
         Step with which the intensity has been recorded
 
+    tracked_compensator : `list[np.ndarray]`, size=n_nodes
+        A record of the compensator of the process evaluated at the jump times.
```

Comment:
Ogata's thinning algorithm entails the computation of the conditional intensities. Hence, the intensity tracker `itr` has an "activation" logic before simulation: if the user is interested in the values of the conditional intensities during simulation (and he wants to access those after the simulation), the values computed during simulation are stored in `itr`. Storing these values is not strictly necessary for the simulation, so the user can decide whether to store them or not, at the price of a slightly slower simulation. 

Instead, the values of the compensators are not needed in Ogata's algorithm. So, the simulation will never compute compensators. If the user is interested in the values of the compensators, the method `store_compensator_values` will compute them at the jump times of the process and store them in the array `ctr`. This always happens after either `timestamps` are set, or a simulation has been performed. So, for compensators  there is no "activation" logic before simulation.  The call to `PP::activate_ctr` happens from within `PP::store_compensator_values`; `PP::activate_ctr` is not meant to be called from anywhere else. It could in fact be a private method of `PP`. I left it as a public method because it resonates with `PP::activate_itr`, which is public,  but  if you prefer I could make `PP::activate_ctr` private. 

I have extended the comments in the code, so hopefully the computation of compensators is clearer.

Reply:
Oh indeed, that is much clearer to me now ! Sorry I missed this logic but it makes full sense, thanks for the very detailed explanation.

Now that I get it better, I am wondering if we could think of a better design. Since it seems that we can compute compensators at any time and that does not affect the PP state, wouldn't it be simpler if we had a function `get_compensator_values` or `compute_compensator_values` that would compute the compensators and return them as a `SArrayDoublePtrList1D` (side remark, if we know what the values length of each compensator will be at the moment of their instantiation, we can use SArrayDoublePtrList1D instead of VArrayDoublePtrList1D) ?

Reply:
We can indeed add methods called `get_compensator_values` to compute the values of compensators at a time $t$ specified by the user; in this way, we would add methods to the `PP` class that would represent the process $t \mapsto \int_0^t \lambda_s ds$, where $\lambda_t$ is the conditional intensity of the point process. This is definitely a good suggestion. 

However, the main objective of this PR is to apply the random time change theorem and draw qq plots, in a way that the user can assess goodness of fit of the data to the point process under examination.  This requires the evaluation of compensators at the timestamps of the realisation, and these values need to be stored. Even if we add `get_compensator_values`, we would have to keep methods to carry out the bespoke evaluations required for the random time change theorem. This is to say that the PR would get larger.

Since the PR is already quite large, I would suggest that we keep the methods `get_compensator_values` for a future PR, and we focus this PR on the random time change theorem and on its usage for QQ plots.  

Generally speaking though, I do recognise the importance of computing compensators at an arbitrary time $t$. For example, my paper [Non-average price impact in order-driven markets](https://arxiv.org/abs/2110.00771) shows how one can use compensators to estimate how stock prices are distorted by execution of large orders.

Reply:
As this was the last item on  discussion, can we merge this PR now? @PhilipDeegan @Mbompr 
I would love to see progress on this :)

Reply:
Yes !!